### PR TITLE
WIP: Support remote hadoop execution on HDI

### PR DESCRIPTION
**Pull request by jjcollinge**
related to #1113

### CODE REVIEW

File: opencga-app/app/scripts/docker/opencga-init/override-yaml.py [hunk 1]
```diff
@@ -59,6 +62,13 @@
         storage_config["cellbase"]["database"]["hosts"].clear()
     storage_config["cellbase"]["database"]["hosts"].insert(i, cellbase_host)
 
+# Inject Hadoop ssh configuration
+if args.hadoop_ssh_host and args.hadoop_ssh_user and args.hadoop_ssh_password:
+    storage_config["storageEngines"][1]["variant"]["options"] = {}
```
Comment (j-coll) [inline opencga-app/app/scripts/docker/opencga-init/override-yaml.py]: Doing this you will drop any other options available in the original configuration file

Comment (jjcollinge) [reply opencga-app/app/scripts/docker/opencga-init/override-yaml.py]: Yeh sorry this was left over from when I thought we were targetted `storage_config["storageEngines"][1]["options"]` which I wasn't sure if always existed. Will fix now, good spot!

**Comment by jjcollinge** [comment]
@j-coll tested this as a deployment. The `opencga` install dir is available on the HDI node at `/tmp/opencga` and the config is set as expected i.e.
```yaml
...
- id: hadoop
  ...
  variant:
      options:
           opencga.hadoop.ssh.host: <ssh-host>
           opencga.hadoop.ssh.user: <ssh-user>
           opencga.hadoop.ssh.password: <ssh-password>
...
```
I've also validated the ssh connection from batch to HDI works as expected and that the `hadoop` cli is avilable. For our deployment this versioned as follows:

```shell
hadoop version
Hadoop 2.7.3.2.6.5.3007-3
Subversion git@github.com:hortonworks/hadoop.git -r 788e99a0a17ade1d18bdeeac7364f3e7e3172ae9
Compiled by jenkins on 2019-01-25T18:26Z
Compiled with protoc 2.5.0
From source with checksum 69e311b4eaa77e633a584514cc586365
This command was run using /usr/hdp/2.6.5.3007-3/hadoop/hadoop-common-2.7.3.2.6.5.3007-3.jar
```
Let me know if you need anything else to get this done

**Comment by jjcollinge** [comment]
As these remote jobs can take hours or even days we need to make sure the config on both the ssh client (Batch node) and the host (HDI) have keep alives set appropriately.

**Comment by jjcollinge** [comment]
merged #1157

**Comment by jjcollinge** [comment]
I have a deployment I think we can test a batch job against in the morning

**Comment by jjcollinge** [comment]
(Copying from slack) : So repor'd the error, with the batch transform completing and 15 minutes later it reporting a connection error. Looking at the pcap - it's the client that terminates the sockets (batch -> mongodb) and mongodb does close those connections immediately. Both batch and mongo have dropped their connections at this point. The Daemon then immedietly moves the job from RUNNING to ERROR, after this the [index-thread] does not log again until 10 mins later when it throws the connection exception. In this state (immediately after it's changed the state of the job) the daemon does not pick up anything from Mongodb (i.e. jobs). However, the pcap on the mongodb server doesn't show any connection issues from the daemonvm. There was constant communication between multiple sockets on the daemonvm and mongodb even after the daemon went into this state but I didn't capture the daemon pcap so can't dig deeper into that. I couldn't grab the java dump of the daemon or batch either :confused: - I copied jmap into the running containers and attempted to attach it to the java process but it failed - I believe because of permissions inside the container - there is no sudo unfortunately :confused:. Guess we'll have to try and repro running the deamon on the host directly next week.

Currently using [nsenter](https://github.com/jpetazzo/nsenter) to caputre a heap dump of the opencga process running on batch.

**Comment by lawrencegripper** [comment]
Notes from standup:

We're looking to merge this as the work to use SSH appears to work correctly. Work will continue on how to resolve the Mongo Timeout issues seen when a transform and load completes on the following item https://github.com/opencb/opencga/issues/1076 as it appears to be the same behavior observed in the Deamon.

**Pull request merged**